core/common/src/files/Paths.kt

```diff
@@ -129,20 +129,46 @@ public fun Path.source(): Source = SystemFileSystem.source(this).buffered()
 @JvmName("sinkDeprecated")
 public fun Path.sink(): Sink = SystemFileSystem.sink(this).buffered()
 
-internal fun removeTrailingSeparators(path: String, sep: Char): String {
+internal fun removeTrailingSeparators(path: String, /* only for testing */ isWindows_: Boolean = isWindows): String {
+    if (isWindows_) {
+        // don't trim the path separator right after the drive name
+        val limit = if (path.length > 1) {
+            if (path[1] == ':') {
+                3
+            } else if (isUnc(path)) {
+                2
+            } else {
+                1
+            }
+        } else {
+            1
+        }
+        return removeTrailingSeparatorsWindws(limit, path)
+    }
+    return removeTrailingSeparatorsUnix(path)
```

Comment:
Is it correct for all native platforms to check on `UnixPathSeparator`?
Shouldn't there be an `SystemPathSeparator` here?

Reply:
Only on Windows both `UnixPathSeparator` and `WindowsPathSeparator` could be used interchangeably (or at least the choice of separator in a particular path won't affect how it is treated by the FS). On all other platforms the only separator is `UnixPathSeparator`.

Reply:
but we only go into the Windows branch in tests, in other cases we always end up here